pavelib/assets.py

```diff
@@ -213,12 +312,105 @@ def compile_coffeescript(*files):
+
+    Command:
+      paver compile_sass --system=cms --themes_dir=/edx/app/edxapp/edx-platform/themes --themes=red-theme,stanford-style
+    Description:
+        compile sass files for cms only for 'red-theme' and 'stanford-style' present in
+        '/edx/app/edxapp/edx-platform/themes'.
+
+    """
+    debug = options.get('debug')
+    force = options.get('force')
+    systems = getattr(options, 'system', ALL_SYSTEMS)
+    themes = getattr(options, 'themes', None)
+    themes_dir = getattr(options, 'themes_dir', None)
+
+    if not themes_dir and themes:
+        # We can not compile a theme sass without knowing the directory that contains the theme.
+        raise ValueError('themes_dir must be provided for compiling theme sass.')
+    else:
+        theme_base_dir = path(themes_dir)
+
+    if isinstance(systems, basestring):
+        systems = systems.split(',')
+    else:
+        systems = systems if isinstance(systems, list) else [systems]
+    if isinstance(themes, basestring):
+        themes = themes.split(',')
+    else:
+        themes = themes if isinstance(themes, list) else [themes]
+
+    timing_info = []
+    dry_run = tasks.environment.dry_run
+    compilation_results = {'success': [], 'failure': []}
+
+    print("\t\tStarted compiling Sass:")
+    for system in systems:
+        for theme in themes:
+            print("Started compiling '{system}' Sass for '{theme}'.".format(system=system, theme=theme or 'system'))
+
+            # Compile sass files
+            is_successful = _compile_sass(
+                system=system,
+                theme=theme_base_dir / theme if theme_base_dir and theme else None,
+                debug=debug,
+                force=force,
+                timing_info=timing_info
+            )
+
+            if is_successful:
+                print("Finished compiling '{system}' Sass for '{theme}'.".format(
+                    system=system, theme=theme or 'system'
+                ))
+
+            compilation_results['success' if is_successful else 'failure'].append('{system} sass for {theme}.'.format(
+                system=system, theme=theme or 'system',
+            ))
+
+    print("\t\tFinished compiling Sass:")
+    if not dry_run:
+        for sass_dir, css_dir, duration in timing_info:
+            print(">> {} -> {} in {}s".format(sass_dir, css_dir, duration))
+
+    if compilation_results['success']:
+        print("\033[92m\n\nSuccessful compilations:\n--- " + "\n--- ".join(compilation_results['success']) + "\033[00m")
+    if compilation_results['failure']:
+        print("\033[91m\n\nFailed compilations:\n--- " + "\n--- ".join(compilation_results['failure']) + "\033[00m")
+
+
+def _compile_sass(system, theme, debug, force, timing_info):
+    """
+    Compile sass files for the given system and theme.
+
+    :param system: system to compile sass for e.g. 'lm', 'cms'
+    :param theme: absolute path of the theme to compile sass for.
+    :param debug:
+    :param force:
```

Comment:
This is missing ``timing_info``. It also isn't using the Google-style for the docstring.